istioctl/cmd/webhook.go

```diff
@@ -0,0 +1,714 @@
+		}
+		if len(opts.mutatingWebhookServiceName) == 0 {
+			return fmt.Errorf("must specify a valid --injection-service")
+		}
+		if len(opts.webhookSecretName) == 0 {
+			return fmt.Errorf("must specify a valid --webhook-secret")
+		}
+	}
+	return nil
+}
+
+type disableCliOptions struct {
+	// The name of the ValidatingWebhookConfiguration to manage
+	validatingWebhookConfigName string
+	// The name of the MutatingWebhookConfiguration to manage
+	mutatingWebhookConfigName string
+	// Whether disable the webhook configuration of Galley
+	disableValidationWebhook bool
+	// Whether disable the webhook configuration of Sidecar Injector
+	disableInjectionWebhook bool
+}
+
+// Validate return nil if no error. Otherwise, return the error.
+func (opts *disableCliOptions) Validate() error {
+	if !opts.disableValidationWebhook && !opts.disableInjectionWebhook {
+		return fmt.Errorf("no webhook configuration to disable")
+	}
+	if opts.disableValidationWebhook && opts.validatingWebhookConfigName == "" {
+		return fmt.Errorf("validating webhook config to disable has empty name")
+	}
+	if opts.disableInjectionWebhook && opts.mutatingWebhookConfigName == "" {
+		return fmt.Errorf("mutating webhook config to disable has empty name")
+	}
+	return nil
+}
+
+type statusCliOptions struct {
+	// Whether display the webhook configuration of Galley
+	validationWebhook bool
+	// Whether display the webhook configuration of Sidecar Injector
+	injectionWebhook bool
+	// The name of the ValidatingWebhookConfiguration to manage
+	validatingWebhookConfigName string
+	// The name of the MutatingWebhookConfiguration to manage
+	mutatingWebhookConfigName string
+}
+
+// Validate return nil if no error. Otherwise, return the error.
+func (opts *statusCliOptions) Validate() error {
+	if !opts.validationWebhook && !opts.injectionWebhook {
+		return fmt.Errorf("no webhook configuration to display")
+	}
+	if opts.validationWebhook && opts.validatingWebhookConfigName == "" {
+		return fmt.Errorf("validating webhook config to display has empty name")
+	}
+	if opts.injectionWebhook && opts.mutatingWebhookConfigName == "" {
+		return fmt.Errorf("mutating webhook config to display has empty name")
+	}
+	return nil
+}
+
+// Webhook command to manage webhook configurations
+func Webhook() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "webhook",
+		Short: "webhook command to manage webhook configurations",
+	}
+
+	cmd.AddCommand(newEnableCmd())
+	cmd.AddCommand(newDisableCmd())
+	cmd.AddCommand(newStatusCmd())
+
+	return cmd
+}
+
+func newEnableCmd() *cobra.Command {
+	opts := &enableCliOptions{}
+	cmd := &cobra.Command{
+		Use:   "enable",
+		Short: "Enable webhook configurations",
+		Example: `
+# Enable the webhook configuration of Galley with the given webhook configuration
+istioctl experimental post-install webhook enable --validation --validation-secret istio.webhook.galley 
+    --namespace istio-system --config-path validatingwebhookconfiguration.yaml
```

Comment:
Example is still using the only `--config-path` option. This needs to be udpated to use the --validation-path and --injection -path.

Reply:
Done.